akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -287,3 +273,25 @@ private[typed] class GuardianActorAdapter[T](_initialBehavior: Behavior[T]) exte
   case object Start
 
 }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[typed] final class ComposedStoppingBehavior[T](lastBehavior: Behavior[T],
+                                                                    stopBehavior: StoppedBehavior[T])
+    extends ExtensibleBehavior[T] {
+  override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] =
+    throw new IllegalStateException("Stopping, should never receieve a message")
+  override def receiveSignal(ctx: TypedActorContext[T], msg: Signal): Behavior[T] = {
+    if (msg != PostStop)
+      throw new IllegalArgumentException(
+        s"The ComposedStoppingBehavior should never receive any other signal than PostStop but got $msg")
+    // first pass the signal to the previous behavior, so that it and potential interceptors
+    // will get the PostStop signal, unless it is deferred, we don't start a behavior while stopping
+    if (!lastBehavior.isInstanceOf[DeferredBehavior[_]])
+      Behavior.interpretSignal(lastBehavior, ctx, PostStop)
+    // and then to the potential stop hook, which can have a call back or not
+    stopBehavior.onPostStop(ctx)
+    Behavior.empty
+  }
+}
```

Comment:
This is essentially the solution: we always pass the signal to the behavior before stopping, before we invoke the PostStop hook (which was before not a callback but a behavior).

This means the interceptors of the previous behavior get the `PostStop` signal, without having to wrap also `StoppedBehavior` with the interceptor.

Previous behavior was to only pass `PostStop` to the previous behavior if there was no callback in the `StoppedBehavior`